### PR TITLE
Up to date conda offline installer PYAPI-2610

### PR DESCRIPTION
**Pull request by rockdreamer**
Many little changes, hopefully for the better:
- Use up to date builders
- remove progress bar in many places, reduces log clutter in CI
- update miniconda to py37_4.9.2
- update conda to latest version before anything else
- Use specific versions that must be kept in sync with cpp-apps-main/wrapping/ccdc/requirements.txt
- remove requirements that don't look so useful
- add a smoke_test.py script that tests whether all imports work or not
- Update  repodata-hotfixes subrepository
- add the condarc to the generated miniconda environment so when users install something, they get conda-forge packages by default

### CODE REVIEW

File: create_offline_installer.py [hunk 1]
```diff
@@ -16,30 +16,38 @@
...
+        'lxml==4.6.3',
+        'numpy==1.20.3', # also used in mercury scripts
         'pytest',
-        'pandas',
-        'py-xgboost',
-        'scikit-learn',
+        'pandas==1.2.4', # also used in mercury scripts
+        'xgboost==1.4.0', # equivalent to py-xgboost, but more used
+        'scikit-learn==0.24.2',
     ]
```
Comment (Alex-AMC) [inline create_offline_installer.py:36]: How does it behave with these exact versions on different platforms? I had issues previously with providing the exact version when switching between Linux and Win.

File: create_offline_installer.py [hunk 2]
```diff
@@ -16,30 +16,38 @@
...
+
     # these packages are required by other scripts that we distribute
+    # Please ensure that these versions are consistent with those in
+    # https://github.com/ccdc-confidential/cpp-apps-main/blob/main/mercury/python-scripts-requirements.txt
     script_pkgs = [
-        'docxtpl', # reports
-        'pockets',
-        'docutils',
-        'pygments',
```
Comment (Alex-AMC) [inline create_offline_installer.py:40]: Is this not required else where?

File: create_offline_installer.py [hunk 3]
```diff
@@ -564,7 +586,7 @@ def build(self):
         print('##[endgroup]')
 
         print('##[group]Pin python version in the installed conda environment', flush=True)
-        self.pin_python_version('3.7')
+        self.pin_python_version()
```
Comment (Alex-AMC) [inline create_offline_installer.py:589]: how come the version isn't pinned anymore?

File: smoke_test.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    from xgboost.compat import XGBoostLabelEncoder
+except:
+    print('Cannot import from xgboost')
+    import_ok = False
+
+if not import_ok:
+    import sys
+    sys.exit(1)
+
+print('All imports worked, huzzah!')
```
Comment (Alex-AMC) [inline smoke_test.py:74]: Do we not need one for SciPy or is it just for the API ?

**Review by Alex-AMC — COMMENTED** [review]
I'm not 100% sure on the azure pipeline ( I have no knowledge of those)

File: create_offline_installer.py (hunk 1 above)
Comment (rockdreamer) [reply create_offline_installer.py:36]: I checked that these versions are carried on all three platforms by conda-forge :)

Comment (rockdreamer) [reply create_offline_installer.py:36]: > How does it behave with these exact versions on different platforms? I had issues previously with providing the exact version when switching between Linux and Win.

I checked that these versions are carried on all three platforms by conda-forge :)
Problem is: the generated miniconda doesn't have a condarc specifying it should pull stuff from conda-forge

File: create_offline_installer.py (hunk 2 above)
Comment (rockdreamer) [reply create_offline_installer.py:40]: I can't find a place where they are used!

File: create_offline_installer.py (hunk 3 above)
Comment (rockdreamer) [reply create_offline_installer.py:589]: it is, see line 472 above :)

File: smoke_test.py (hunk 4 above)
Comment (rockdreamer) [reply smoke_test.py:74]: The goal of this offline repository is to have all the dependencies used by the api plus everything that is used by the mercury and hermes scripts we ship. If scipy is used in the api, I should add a check for it.

**Review by siong-chin — APPROVED** [review]

File: smoke_test.py (hunk 4 above)
Comment (Alex-AMC) [reply smoke_test.py:74]: I've just done a search for scipy and it seems to crop up in 10 different files 😅

https://github.com/ccdc-confidential/cpp-apps-main/search?q=import+scipy

Comment (rockdreamer) [reply smoke_test.py:80]: Added

**Review by Alex-AMC — APPROVED** [review]

**Pull request merged**